AgriCodeHub/dairy-django-backend
# Pull Request: Create Pregnancy Model for Dairy Django Backend

## Issue Description:
Resolves #11

## Changes:
I have implemented the Pregnancy model as per the requirements outlined in the issue.

### Pregnancy Model Definition:
- Added attributes `cow`, `start_date`, `date_of_calving`, `pregnancy_status`, `pregnancy_notes`, `calving_notes`, `pregnancy_scan_date`, `pregnancy_failed_date`, and `pregnancy_outcome` to efficiently manage and record pregnancy-related information for cows.
  
### Methods:
- Implemented the `pregnancy_duration` method, which returns the number of days since the inception of pregnancy.
- Implemented the `due_date` method, which returns the most the estimated calving date of the cow.

### Validation:
- Implemented validation checks before saving the pregnancy record to ensure data integrity.
  - Validate the age of the associated cow at the start of pregnancy.
  - Check the current pregnancy status and availability status of the cow.
  - Ensure logical relationships between pregnancy-related dates.
  - Validate the pregnancy outcome based on the pregnancy status.

## Discussion Points:
I would like to discuss the following points:

1. **Attributes and Methods:**
   - Review and provide feedback on the attributes and methods added to the Pregnancy model.

2. **Validation Checks:**
   - Discuss the implemented validation checks to ensure they align with the project's requirements.

3. **Additional Features or Modifications:**
   - Open to suggestions for any additional features or modifications needed for the Pregnancy model.

## Testing:
I have tested the Pregnancy model in various scenarios to ensure the correctness of the implementation.

## Documentation:
Updated the project documentation to include information about the new Pregnancy model and its methods.

## Acknowledgment of Feedback:
I am open to feedback and willing to address any concerns or make further improvements based on the reviewers' suggestions.

## Next Steps:
Once the changes are reviewed and approved, I will proceed with merging the pull request into the main branch.
